Nickvision.Aura/Network/NetworkMonitor.cs

```diff
@@ -11,30 +11,13 @@ namespace Nickvision.Aura.Network;
 /// </summary>
 public class NetworkMonitor : IDisposable
 {
-    public event Action<NetworkState>? _stateChanged;
-    
-    /// <summary>
-    /// Occurs when network state is changed
-    /// </summary>
-    public event Action<NetworkState> StateChanged
-    {
-        add
-        {
-            _stateChanged += value;
-            Task.Run(async () => _stateChanged.Invoke(await GetStateAsync()));
-        }
-
-        remove
-        {
-            _stateChanged -= value;
-        }
-    }
-    
     private bool _disposed;
     private Connection? _dbusConnection;
     private INetworkManager? _networkManager;
     private string[]? _networkAddresses;
     
+    private event Action<NetworkState>? _stateChanged;
```

Comment:
`EventHandler<NetworkState>?`

Reply:
events should also be with `EventHandler` not `Action`.

Reply:
Is there a reason for that? We don't really need a sender, so Action works in our case

Reply:
I don't know of any technically differences between EventHandler vs Action as function pointers, but when using C#'s `event` system, `EventHandlers` should be used as that's the practice. Cause then with inheriting events and custom EventHandler types it just keeps everything playing nice.